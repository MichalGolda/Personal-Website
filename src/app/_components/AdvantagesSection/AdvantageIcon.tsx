import Image from "next/image";

export type AdvantageIconProps = {
  src: string;
  alt: string;
};

const AdvantageIcon = ({ src, alt }: AdvantageIconProps) => (
  <Image
    src={src}
    alt={alt}
    width={64}
    height={64}
    className="w-[48px] lg:w-[64px]"
  />
);

export default AdvantageIcon;
