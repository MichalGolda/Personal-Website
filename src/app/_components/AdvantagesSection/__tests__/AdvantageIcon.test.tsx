import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import AdvantageIcon from "../AdvantageIcon";

describe("AdvantageIcon component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <AdvantageIcon src="/test.webp" alt="test alt" />
    );

    const image = container.firstChild;

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute(
      "src",
      "/_next/image?url=%2Ftest.webp&w=128&q=75"
    );
    expect(image).toHaveAttribute("alt", "test alt");
    expect(image).toHaveAttribute("width", "64");
    expect(image).toHaveAttribute("height", "64");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <AdvantageIcon src="/test.webp" alt="test alt" />
    );

    expect(container).toMatchSnapshot();
  });
});
