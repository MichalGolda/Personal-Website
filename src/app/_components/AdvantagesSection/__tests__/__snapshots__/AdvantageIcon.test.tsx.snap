// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvantageIcon component renders unchanged 1`] = `
<div>
  <img
    alt="test alt"
    class="w-[48px] lg:w-[64px]"
    data-nimg="1"
    decoding="async"
    height="64"
    loading="lazy"
    src="/_next/image?url=%2Ftest.webp&w=128&q=75"
    srcset="/_next/image?url=%2Ftest.webp&w=64&q=75 1x, /_next/image?url=%2Ftest.webp&w=128&q=75 2x"
    style="color: transparent;"
    width="64"
  />
</div>
`;
